fixtures/_fixtures/monkeypatch.py

```diff
@@ -17,27 +17,70 @@
     'MonkeyPatch'
     ]
 
+import functools
 import sys
 import types
 
 from fixtures import Fixture
 
 
-def _setattr(obj, name, value):
-    """Handle some corner cases when calling setattr.
+_class_types = (type, )
+if getattr(types, 'ClassType', None):
+    # Python 2 has multiple types of classes.
+    _class_types = _class_types + (types.ClassType,)
 
-    setattr transforms a function into instancemethod, so where appropriate
-    value needs to be wrapped with staticmethod().
+
+def _coerce_values(obj, name, new_value, sentinel):
+    """Handle value coercion for static and classmethods.
+
+    setattr transforms a function into an instancemethod when set on a class.
+    This checks if the attribute to be replaced is either and wraps new_value
```

Comment:
is either what? The previous sentence contains three possible nouns (function, instancemethod, class). This doesn't make any sense to me. Can you re-word please?